release-notes/5.0/api-diff/netcoreapp3.1/.Net/5.0_System.Diagnostics.md

```diff
@@ -0,0 +1,140 @@
++    }
++    public enum ActivitySamplingResult {
++        AllData = 2,
++        AllDataAndRecorded = 3,
++        None = 0,
++        PropagationData = 1,
++    }
++    public sealed class ActivitySource : IDisposable {
++        public ActivitySource(string name, string version = "");
++        public string Name { get; }
++        public string Version { get; }
++        public static void AddActivityListener(ActivityListener listener);
++        public void Dispose();
++        public bool HasListeners();
++        public Activity StartActivity(string name, ActivityKind kind = ActivityKind.Internal);
++        public Activity StartActivity(string name, ActivityKind kind, ActivityContext parentContext, IEnumerable<KeyValuePair<string, object>> tags = null, IEnumerable<ActivityLink> links = null, DateTimeOffset startTime = default(DateTimeOffset));
++        public Activity StartActivity(string name, ActivityKind kind, string parentId, IEnumerable<KeyValuePair<string, object>> tags = null, IEnumerable<ActivityLink> links = null, DateTimeOffset startTime = default(DateTimeOffset));
++    }
++    public class ActivityTagsCollection : ICollection<KeyValuePair<string, object>>, IDictionary<string, object>, IEnumerable, IEnumerable<KeyValuePair<string, object>> {
++        public ActivityTagsCollection();
++        public ActivityTagsCollection(IEnumerable<KeyValuePair<string, object>> list);
++        public int Count { get; }
++        public bool IsReadOnly { get; }
++        public ICollection<string> Keys { get; }
++        public object this[string key] { get; set; }
++        public ICollection<object> Values { get; }
++        public void Add(KeyValuePair<string, object> item);
++        public void Add(string key, object value);
++        public void Clear();
++        public bool Contains(KeyValuePair<string, object> item);
++        public bool ContainsKey(string key);
++        public void CopyTo(KeyValuePair<string, object>[] array, int arrayIndex);
++        public ActivityTagsCollection.Enumerator GetEnumerator();
++        public bool Remove(KeyValuePair<string, object> item);
++        public bool Remove(string key);
++        IEnumerator<KeyValuePair<string, object>> System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<System.String,System.Object>>.GetEnumerator();
++        IEnumerator System.Collections.IEnumerable.GetEnumerator();
++        public bool TryGetValue(string key, out object value);
++        public struct Enumerator : IDisposable, IEnumerator, IEnumerator<KeyValuePair<string, object>> {
++            public KeyValuePair<string, object> Current { get; }
++            object System.Collections.IEnumerator.Current { get; }
++            public void Dispose();
++            public bool MoveNext();
++            void System.Collections.IEnumerator.Reset();
++        }
++    }
+-    public class Process : Component {
++    public class Process : Component, IDisposable {
```

Comment:
Process always implemented IDisposable because Component implemented it, so this is a strange change to show.